### DISCUSSION TITLE
How to aviod parsing variable with BaseSettings

### DISCUSSION THREAD

**Discussion by SnakeOilSalesman** [pydantic/pydantic#4440] [Question]
Hello.

I'd like to use settings classes, that are populated with default values and with ones loaded from the environment in the same class. Not sure what's the best way.

Here's example:
```Python
class Settings(BaseSettings):
    default_only_value: int = 123  # this one expected to be populated with value 123 and avoid overloading, ignoring env
    env_value: str  #  this one expected to be populated from environment
```
I'm worried about potential overload of the default value of `Settings.default_only_value`, when there is an env variable with the same name.

Have tried this:
```Python
class Settings(BaseSettings):
    default_only_value: int = Field(123, const=True)
    env_value: str
```
It'll produce `ValidationError` if there's a variable named `default_only_value` in my environ. And I want simply ignore it.

Here's another approach:
```Python
class Settings(BaseSettings):
    default_only_value: int = Field(123, env='')
    env_value: str
```
And it works as expected, even if there are few fields having `env=''`, but I'm not sure if it's safe or right way to do it.

**Comment by old-syniex** [comment]
Add a validator that will return always 123 for the field

**Reply by SnakeOilSalesman** [reply]
> Add a validator that will return always 123 for the field

Hey! Thanks for your reply.

But this is against the idea of using default values specified with `Field`.
I'd like to stay closer to the basic usage. And in real scenario it'll require to add a bunch of validators for each field or root validator with mapping.

**Comment by samuelcolvin** [comment]
Implement a custom [`customise_sources`](https://github.com/pydantic/pydantic/blob/f8575aa3793c565cf006979c60ad62312406d5b5/pydantic/env_settings.py#L71) method.

**Reply by SnakeOilSalesman** [reply]
Am I getting your idea right?

```Python
    class Config:
        @classmethod
        def customise_sources(
            cls,
            init_settings: SettingsSourceCallable,
            env_settings: SettingsSourceCallable,
            file_secret_settings: SettingsSourceCallable,
        ) -> tuple[SettingsSourceCallable, ...]:
            return init_settings, file_secret_settings
```

Omitting `env_settings`?

If so, then it's not exactly what I'm after. I want somehow mark specific fields to exclude them from parsing of environ variables. Ideally, by passing some argument to `Field`. To make it more explicit.

Using `const=True` looks like what I'm searching for, but, as I understand, it's impossible to exclude a specific field from validation.

**Reply by samuelcolvin** [reply]
Yes, that's the idea, but still return `env_settings`, with some items removed.
